gravity_upgrade.md

```diff
@@ -0,0 +1,101 @@
+
+The Gravity upgrade process handles both of these mentioned layers in the following manner:
+
+1. A new version of an application and container images are downloaded and stored inside the cluster and transferred onto a cluster node. 
+
+2. Gravity uses k8s rolling update mechanism to perform the update.
+
+3. If applicable, before or after the upgrade custom update hooks can be used to perform application specific actions.
+
+Upgrades can be triggered both via the Gravity Hub (Ops Center) or from the CLI. Below we will go over the upgrade via the CLI to give some detail to the above steps mentioned.
+
+### Step 1: Providing new version of application
+
+As mentioned the first step in the upgrade process is to import your new application bundle onto a Gravity cluster and transfer it onto a cluster nodes to execute the upgrade command.
+
+Upgrades can be handled both in online and offline environments.
+
+#### Online Gravity Upgrade
+
+In the case a Gravity cluster is connected to the Gravity Hub (Ops Center) and it has remote support turned on, you can download an updated application bundle directly from the Ops Center. 
+
+Using the `gravity update download` you can check and download any updates
+
+#### Offline Gravity Upgrade
+
+In the case where a Gravity cluster is offline the new version of the application bundle will need to be copied over to one of the application cluster nodes that is accessible to the other nodes.
+
+In order to upload to the new version you will need to extract the tarball and launch the `upload` script.
+
+### Step 2: Performing Gravity Upgrade
+
+As mentioned a Gravity upgrade can be triggered via the Gravity Hub (Ops Center) or from the CLI. In order to trigger the upgrade from the CLI you can extra the application bundle tarball to a directory and inside the execute the `upgrade` script.
+
+```
+tar xf application-bundle.tar
+ls -lh
+sudo ./gravity upgrade
+```
+#### Automatic Upgrade
+
+Following executing the upgrade command following the automatic upgrade process, the following steps will take place:
+
+1. Upgrade agent gets deployed on each cluster node as a `systemd` unit called `gravity-agent.service`
+
+`./gravity agent deploy`
+
+2. Phases listed in the upgrade plan are executed by the agents.
+
+`./gravity plan execute --phase=<phase>`
+
+3. Upon completion the agents are shut down.
+
+`./gravity agent shutdown`
+
+#### Manual Upgrade
+
+Another option for a Gravity upgrade is a manual upgrade which we will cover by stepping through an example later in this workshop.
+
+In order to execute a manual upgrade the operation will be started by adding `--manual | -m flag` with the upgrade command:
+
+`sudo ./gravity upgrade --manual`
+
+To ensure version compatibility the manual upgrade process requires the Gravity binary to be included in the upgrade tarball.
```

Comment:
I would rephrase this to say that to ensure version compatibility, all upgrade related commands (agent deployment, phase execution/rollback, etc.) need to be executed using the gravity binary included in the upgrade tarball.